hw/ip/otbn/doc/_index.md

```diff
@@ -522,45 +522,66 @@ The LFSR state is advanced every cycle when OTBN is running.
+
+1. The currently running operation is terminated, similar to the way an {{< otbnInsnRef "ECALL" >}} instruction [is executed](#writing-otbn-applications-ecall):
+   - No more instructions are fetched or executed.
+   - An [internal state secure wipe](#design-details-secure-wipe-internal) is performed.
+   - The {{< regref "ERR_BITS" >}} register is set to a non-zero value describing the error.
+   - The current operation is marked as completed by setting {{< regref "INTR_STATE.done" >}} and clearing {{< regref "STATUS.busy" >}}.
+2. A [recoverable alert]({{< relref "#alerts" >}}) is raised.
+
+The host software can start another operation on OTBN after a recoverable error was detected.
+
+### Fatal Errors {#design-details-fatal-errors}
+
+Fatal errors are generally seen as a sign of an intrusion, resulting in more drastic measures to protect the secrets stored within OTBN.
+Fatal errors can occur at any time.
+
+The following actions are taken when OTBN detects a fatal error:
+
+1. A [secure wipe of the data](#design-details-secure-wipe-dmem) and [instruction memories](#design-details-secure-wipe-imem) is initiated.
+2. If OTBN is busy, as indicated by {{< regref "STATUS.busy" >}},
+   then the currently running operations is terminated, similar to the way an {{< otbnInsnRef "ECALL" >}} instruction [is executed](#writing-otbn-applications-ecall):
+   - No more instructions are fetched or executed.
+   - An [internal state secure wipe](#design-details-secure-wipe-internal) is performed.
+   - The {{< regref "ERR_BITS" >}} register is set to a non-zero value describing the error.
+   - The current operation is marked as completed by setting {{< regref "INTR_STATE.done" >}} and clearing {{< regref "STATUS.busy" >}}.
+3. A [fatal alert]({{< relref "#alerts" >}}) is raised.
 
 Note that OTBN can detect some errors even when it isn't running.
 One example of this is an error caused by an integrity error when reading or writing OTBN's memories over the bus.
 In this case, the {{< regref "ERR_BITS" >}} register will not change.
 This avoids race conditions with the host processor's error handling software.
-However, every error that OTBN detects when it isn't running causes a fatal alert.
+However, every error that OTBN detects when it isn't running is fatal.
 This means that the cause will be reflected in {{< regref "FATAL_ALERT_CAUSE" >}}, as described below in [Alerts]({{< relref "#alerts" >}}).
 This way, no alert is generated without setting an error code somewhere.
 
-<div class="bd-callout bd-callout-warning">
-  <h5>TODO</h5>
-
-  When the implementation is finished, document more precisely how OTBN stops on error.
-  Can we claim to cancel all register and memory writes in that cycle?
-  Is there a way for that to make sense for errors that aren't related to a particular instruction (e.g. shadow register mis-matches; FSM glitches)?
-
-  Also, once it is decided, document behaviour on a bus access when OTBN is running.
```

Comment:
I don't think we've done this bit yet?

Reply:
No, it's not yet implemented, pending https://github.com/lowRISC/opentitan/issues/2696. Since we already have a tracking issue for it, and we don't really track all missing features in the spec text as TODO comments, I removed it.